sample.js

```diff
@@ -0,0 +1,80 @@
+import debug from 'debug'
+import { randomInt, randomBigInt, sleep } from './utils.js'
+
+/** @typedef {{ cid: string, peer: string }} Sample */
+
+const log = debug('checkup:sample')
+/**
+ * 8k max request length to cluster for statusAll, we hit this at around 126 CIDs
+ * http://nginx.org/en/docs/http/ngx_http_core_module.html#large_client_header_buffers
+ */
+const MAX_CLUSTER_STATUS_CIDS = 120
+
+/**
+ * @param {import('pg').Client} db
+ */
+async function fetchMinMaxUploadId (db) {
+  const { rows } = await db.query('SELECT MIN(id), MAX(id) FROM upload')
+  if (!rows.length) throw new Error('no rows returned fetching min/max ID')
+  return { min: BigInt(rows[0].min), max: BigInt(rows[0].max) }
+}
+
+/**
+ * @param {import('pg').Client} db
+ * @param {bigint} id
+ * @returns {{ source_cid: string }|undefined}
+ */
+async function fetchUploadById (db, id) {
+  const { rows } = await db.query('SELECT source_cid FROM upload WHERE id = $1', [id.toString()])
```

Comment:
That would be a useful metric...I'm not sure it fits with the purpose of _this_ library. It's more analytical than verifyable if that makes sense. It's also a little bit hand wavy since the same CID may be uploaded/pinned by multiple people.